includes/class-wc-rest-authentication.php

```diff
@@ -66,6 +67,102 @@ protected function is_request_to_rest_api() {
 		return apply_filters( 'woocommerce_rest_is_request_to_rest_api', $woocommerce || $third_party );
 	}
 
+	/**
+	 * Check if this is a request to WCCOM Site REST API.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return bool
+	 */
+	protected function is_request_to_wccom_site_rest_api() {
+		$request_uri = add_query_arg( array() );
+		$rest_prefix = trailingslashit( rest_get_url_prefix() );
+		$request_uri = esc_url_raw( wp_unslash( $request_uri ) );
+
+		return false !== strpos( $request_uri, $rest_prefix . 'wccom-site/' );
+	}
+
+	/**
+	 * Verify WooCommerce.com request from a given body and signature request.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @param string $body                Request body.
+	 * @param string $signature           Request signature found in
+	 *                                    X-Woo-Signature header.
+	 * @param string $access_token_secret Access token secret for this site.
+	 *
+	 * @return bool
+	 */
+	protected function verify_wccom_request( $body, $signature, $access_token_secret ) {
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$data = array(
+			'host'        => $_SERVER['HTTP_HOST'],
+			'request_uri' => $_SERVER['REQUEST_URI'],
+			'method'      => strtoupper( $_SERVER['REQUEST_METHOD'] ),
+		);
+		// phpcs:enable
+
+		if ( ! empty( $body ) ) {
+			$data['body'] = $body;
+		}
+
+		$expected_signature = hash_hmac( 'sha256', wp_json_encode( $data ), $access_token_secret );
+
+		return hash_equals( $expected_signature, $signature );
+	}
+
+	/**
+	 * Authenticate WooCommerce.com request.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @param int|false $user_id User ID.
+	 *
+	 * @return int|false
+	 */
+	public function authenticate_wccom( $user_id ) {
+		if ( ! empty( $user_id ) || ! $this->is_request_to_wccom_site_rest_api() ) {
+			return $user_id;
+		}
+
+		if ( empty( $_SERVER['HTTP_AUTHORIZATION'] ) ) {
```

Comment:
This is not present in `$_SERVER` in all servers, that's why we have a `$this->get_authorization_header()` method, so needs to replace it.

Reply:
Added helper method from `get_authorization_header()` in https://github.com/woocommerce/woocommerce/pull/24123/commits/c8812a205bfe249fa6ddb9ab7a209425de73d081. It's copied as is from `class-wc-rest-authentication.php` instead of depending on the class.